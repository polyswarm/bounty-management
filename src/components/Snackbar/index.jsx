// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty Manager imports

class Snackbar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { message } = this.props;
    return(
      <p className='Snackbar'>
        {message}
      </p>
    );
  }

}
Snackbar.proptypes = {
  message: PropTypes.string
};
export default Snackbar;