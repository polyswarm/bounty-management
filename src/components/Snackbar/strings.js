const strings = {
  dismiss: 'DISMISS'
};
export default strings;