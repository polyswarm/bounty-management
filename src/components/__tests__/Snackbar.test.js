import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Snackbar from '../Snackbar';

it('renders without crashing', () => {
  const wrapper = render(<Snackbar />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('shows the text given as the message property', () => {
  const message = 'Show this error?';
  const wrapper = mount(<Snackbar message={message}/>);

  expect(wrapper.find('p').text()).toEqual('Show this error?');
});

it('calls onDismiss when dismiss is clicked', () => {
  const onDismiss = jest.fn();
  const message = 'Show this error?';
  const wrapper = mount(<Snackbar message={message} onDismiss={onDismiss}/>);

  wrapper.find('button').simulate('click');

  expect(onDismiss).toHaveBeenCalledTimes(1);
});
