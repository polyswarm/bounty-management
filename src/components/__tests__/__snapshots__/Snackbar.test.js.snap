// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Snackbar"
>
  <p>
    
  </p>
  <button
    class="Button flat"
  >
    DISMISS
  </button>
</div>
`;
